packages/reakit/src/Button/__examples__/ButtonWithTootltip/index.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from "react";
+import { Button } from "reakit/Button";
+import { Tooltip, TooltipReference, useTooltipState } from "reakit/Tooltip";
+import { VisuallyHidden } from "reakit/VisuallyHidden";
+
+type Props = {
+  isLocked: boolean;
+  onClick: () => void;
+};
+
+function LockButton({ isLocked, ...rest }: Props) {
+  const tooltip = useTooltipState();
+  return (
+    <>
+      <TooltipReference {...rest} {...tooltip} as={Button}>
```

Comment:
Let's push the pattern of always putting the state object first so people can override state props if they want.
```suggestion
function LockButton({ isLocked, ...props }: Props) {
  const tooltip = useTooltipState();
  return (
    <>
      <TooltipReference {...tooltip} {...props} as={Button}>
```

Reply:
Actually, nevermind! In this case I think we should indeed put `{...props}` first as we're only accepting HTML props.

Reply:
I was not decided about it too :)
But, yes, I rename it.